src/select.js

```diff
@@ -167,12 +167,11 @@ const CanopySelect = React.createClass({
 		var i = this.getIndexFromString(searchString);
 
 		if(i > -1) {
-			this.setState({
-				selectedIndex: i,
-				//			dialogDisplayed: true
-			});
+			this.selectItem(i);
```

Comment:
If the value has been highlighted, it has been changed, calling `selectItem` will triggers an `onChange` event.